#
# Submitted-By: Robert Schwebel, 2006-05-26
# Committed-By: Robert Schwebel
#
# Error:
#
# bashline.c:1936: undefined reference to `history_expand_line_internal'
#
# Description:
#
# In bashline.c, history_expand_line_internal is used in a block where
# BANG_HISTORY is not declared. So we define it unconditionally here.
#
# State:
# 
# unknown
#

diff -urN bash-3.0-orig/bashline.c bash-3.0/bashline.c
--- bash-3.0-orig/bashline.c	2004-07-06 05:22:12.000000000 +0200
+++ bash-3.0/bashline.c	2006-05-26 09:17:11.000000000 +0200
@@ -87,11 +87,10 @@
 static int bash_ignore_filenames __P((char **));
 static int bash_ignore_everything __P((char **));
 
-#if defined (BANG_HISTORY)
 static char *history_expand_line_internal __P((char *));
 static int history_expand_line __P((int, int));
 static int tcsh_magic_space __P((int, int));
-#endif /* BANG_HISTORY */
+
 #ifdef ALIAS
 static int alias_expand_line __P((int, int));
 #endif
@@ -152,9 +151,7 @@
 #endif
 
 /* Variables used here but defined in other files. */
-#if defined (BANG_HISTORY)
 extern int hist_verify;
-#endif
 
 extern int current_command_line_count, last_command_exit_value;
 extern int posixly_correct, no_symbolic_links;
@@ -1732,7 +1729,6 @@
 
 /* Functions to perform history and alias expansions on the current line. */
 
-#if defined (BANG_HISTORY)
 /* Perform history expansion on the current line.  If no history expansion
    is done, pre_process_line() returns what it was passed, so we need to
    allocate a new line here. */
@@ -1750,7 +1746,6 @@
 
   return (new_line == line) ? savestring (line) : new_line;
 }
-#endif
 
 /* There was an error in expansion.  Let the preprocessor print
    the error here. */
